lib/rules/sort-vars.js

```diff
@@ -27,29 +27,68 @@ module.exports = {
                 },
                 additionalProperties: false
             }
-        ]
+        ],
+
+        fixable: "code"
     },
 
     create(context) {
 
         const configuration = context.options[0] || {},
-            ignoreCase = configuration.ignoreCase || false;
+            ignoreCase = configuration.ignoreCase || false,
+            sourceCode = context.getSourceCode();
 
         return {
             VariableDeclaration(node) {
                 const idDeclarations = node.declarations.filter(decl => decl.id.type === "Identifier");
+                const getSortableName = ignoreCase ? decl => decl.id.name.toLowerCase() : decl => decl.id.name;
+                const unfixable = idDeclarations.some(decl => decl.init !== null && decl.init.type !== "Literal");
+                let fixed = false;
 
                 idDeclarations.slice(1).reduce((memo, decl) => {
-                    let lastVariableName = memo.id.name,
-                        currenVariableName = decl.id.name;
+                    const lastVariableName = getSortableName(memo),
+                        currentVariableName = getSortableName(decl);
 
-                    if (ignoreCase) {
-                        lastVariableName = lastVariableName.toLowerCase();
-                        currenVariableName = currenVariableName.toLowerCase();
-                    }
+                    if (currentVariableName < lastVariableName) {
+                        if (unfixable || fixed) {
```

Comment:
Nitpick: maybe it make more sense to place this check to fixer function. (returning null means do not fix.)

Reply:
Thanks! I did not know about returning null in a fixer. I will change this.